Detectors/Passive/src/Pipe.cxx

```diff
@@ -8,99 +8,2921 @@
+  //  Pos 7 - Minimised Flange:                                LHCVFX_P0025
+  //  Pos 6 - Standard Flange:                                 STDVFUHV0009
+  //  Pos 8 - Bellow:                                          LHCVBX__0001
+  //
+  //  Absolute z-coordinates -82.0 - 400.0 cm
+  //  Total length:                                          482.0 cm
+  //  It consists of 3 main parts:
+  //  CP/2 The flange on the non-absorber side:               36.5 cm
+  //  CP/1 The central Be pipe:                              405.0 cm
+  //  CP/3 The double-bellow and flange on the absorber side: 40.5 cm
+
+  //
+  /*
+  //  Starting position in z
+  const Float_t kCPz0      = -400.0;
+  //  Length of the CP/1 section
+  const Float_t kCP1Length =  405.0;
+  //  Length of the CP/2 section
+  const Float_t kCP2Length =   36.5;
+  //  Length of the CP/3 section
+  const Float_t kCP3Length =   40.5;
+  //  Position of the CP/2 section
+  //    const Float_t kCP2pos    = kCPz0 + kCP2Length / 2.;
+  //  Position of the CP/3 section
+  const Float_t kCP3pos    = kCPz0 + kCP2Length + kCP1Length + kCP3Length/2.;
+  */
+
+  //////////////////// NEW BEAM PIPE GEOMETRY FOR MuonForwardTracker ,
+  // Authors: F. Manso, R. Tieulent
+  // Drawings from C. Gargiulo :
+  // \\cern.ch\dfs\Workspaces\c\cgargiul\EXPERIMENT\ALICE\ALICE_MECHANICS\ALICE_DATA_PACKAGE\IN\DETECTORS\ITS_UPGRADE\1-DESIGN\3D_cad_model\R14_20140311_ALI\
+  //
+  //------------------- Pipe version 4.7 March 2014 -----------------------------
+
+  TGeoVolumeAssembly* beamPipeCsideSection = new TGeoVolumeAssembly("BeamPipeCsideSection");
+
+  Float_t fBeryliumSectionOuterRadius = 1.9;
+  Float_t fBeryliumSectionZmax = 44.4;
+  Float_t fBeryliumSectionZmin = -44.4;
+  Float_t fBeryliumSectionThickness = 0.08;
+
+  Float_t fBellowSectionOuterRadius = 2.15;
+  Float_t fCSideBPSOuterRadius = 2.22;
+  Float_t fCSideBPSWallThickness = 0.15;
+  Float_t fBellowSectionZmax = -55.35;
+  Float_t fBellowOuterRadius = 2.8;
+  Float_t fFirstConeAngle = 15. * TMath::DegToRad();
+  Float_t fChangeThicknessAngle = 45. * TMath::DegToRad();
+  Float_t fCSideBPSLength = 3.53;
+  //  Float_t fCSideBPSLength             = 3.53+1.52;
+  Float_t fDzFirstCone = (fCSideBPSOuterRadius - fBeryliumSectionOuterRadius) / TMath::Tan(fFirstConeAngle);
+  Float_t fReduceThicknessPartAfterBPSLength = 1.52;
+  //  Float_t fReduceThicknessPartAfterBPSLength = 0.;
+  Float_t fThinPartBeforeBellowLength = 1.025;
+
+  Float_t fDistanceBetweenBellows = 2.5;
+
+  Float_t fAdaptConeZmax = -77.43;
+  Float_t fAdaptConeZmin = -80.6;
+  Float_t fAdaptConeRmax = 3.0;
+  Float_t fFlangeRmax = 4.3;
+  Float_t fFlangeLength = 1.4;
+
+  Float_t fBellowPlieRadius = 0.17;    // radius of bellow plies
+  Float_t fBellowPlieThickness = 0.03; // Thickness of bellow plies 300 microns
+  Int_t fNBellowConvolutions = 7;
+
+  // If user set Rmax=0/Thick=0 use defaults, else use user input
+  if (mBePipeRmax > 0.)
+    fBeryliumSectionOuterRadius = mBePipeRmax;
```

Comment:
please check if there is a potential problem here: `fDzFirstCone` is calculated from the default and used without adjustment after setting the user define outer radius